src/metabase/warehouses/models/database.clj

```diff
@@ -60,10 +60,7 @@
 (doto :model/Database
   (derive :metabase/model)
   (derive :hook/timestamped?)
-  ;; Deliberately **not** deriving from `:hook/entity-id` because we should not be randomizing the `entity_id`s on
```

Comment:
I think this comment is wrong. If it's not, please explain. Sync cannot create databases - ever. Databases are manually created by admins. So yes, we cannot set random entity_id on tables and fields on creation, but we can - and should - on databases.